tasks.py

```diff
@@ -42,6 +29,105 @@ def get_loss(self, predictions, labels):
     def calculate_accuracy(self, predictions, labels):
         raise NotImplementedError
 
+    def get_name(self):
+        return self.NAME
+
+
+class TaskSamplerIter(object):
+    """Iterator class used by TaskSampler."""
+    def __init__(self, task_iters):
+        self.task_iters = [iter(ti) for ti in task_iters]
+        self._len_tasks_called = sum([len(task_iter) for task_iter in task_iters])
+        self.task_indexes = list(range(len(task_iters)))
+        task_num_examples = [len(task_iter) for task_iter in task_iters]
+        total_num_examples = sum(task_num_examples)
```

Comment:
Are task_num_examples and total_num_examples needed?
Or does self._len_tasks_called replace their functionality?